File: src/main/java/io/github/chaosdave34/kitpvp/abilities/impl/poseidon/StormAbility.java
package io.github.chaosdave34.kitpvp.abilities.impl.poseidon;

import io.github.chaosdave34.kitpvp.abilities.Ability;

import net.kyori.adventure.text.Component;
import org.bukkit.World;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class StormAbility extends Ability {
    public StormAbility() {
        super("storm", "Storm", Type.SNEAK_RIGHT_CLICK, 120);
    }

    @Override
    public @NotNull List<Component> getDescription() {
        return createSimpleDescription("Gather a storm for 30 seconds.");
    }

    @Override
    public boolean onAbility(Player p) {
        World world = p.getWorld();
        world.setStorm(true);
        world.setWeatherDuration(600);
        world.setThundering(true);
        world.setThunderDuration(600);
        return true;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/abilities/impl/poseidon/StormAbility.kt
package io.github.chaosdave34.kitpvp.abilities.impl.poseidon

import io.github.chaosdave34.kitpvp.abilities.Ability
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player

class StormAbility : Ability("storm", "Storm", Type.SNEAK_RIGHT_CLICK, 120) {
    override fun getDescription(): List<Component> = createSimpleDescription("Gather a storm for 45 seconds.")

    override fun onAbility(player: Player): Boolean {
        val world = player.world
        world.setStorm(true)
        world.weatherDuration = 45 * 20
        world.isThundering = true
        world.thunderDuration = 45 * 20
        return true
    }
}